### PR TITLE
♻️🔨 Is3517/refactor service io and diagnostics tool concept

### LINKED ISSUE #3517: Refactor catalog to use utils.service_io
Centralize schema transformations in ``utils.service_io``

- ``packages/models-library/src/models_library/utils/services_io.py``

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ♻️ Moves service i/o utilities to a common place: ``models_library.utils.services_io``
- 🔨 Tools to diagnose image metadata in deployed registries 
- ♻️ cleanup ``pytest_simcore.environment_configs`` fixtures
    - can be used as an executable or as an util in testing 
    - why in ``pytest_simcore``?  Because it can be imported by all testing

### 🔨 Tools to diagnose image metadata in deployed registries
- *Motivation*: Frequently I want to validate actual production data (e.g. database, registry...) against a new invariant that i am developing (e.g. model, constraints, etc).  
- *Rationale*:  Instead of creating a separate script, let's **reuse** test infrastructure as a diagnostics tool. Tests used in diagnostics are marked with  ``@pytest.mark.diagnostics``. If in the diagnostics we find edge/interesting cases, we can version the data under ``test/data``.
- We demonstrate this concept by downloading **the metadata of all images** in a registry and validate it against i/o pydantic models and json-schema-validators. These are the steps:
      -  configure the registry under test in ``.env``
      -  ``make pull_test_data`` to download all metadata as files
      -  ``pytest --vv -m diagnostics`` to run diagnostics on the pulled data

Running diagnostics against master registry already identified a
```cmd
================================================================== short test summary info ===================================================================
FAILED tests/test_utils_service_io.py::test_against_service_metadata_configs[sleeper/metadata-0.0.1.json] - pydantic.error_wrappers.ValidationError: 1 validation error for ParsingModel[dict[models_library.services.ConstrainedStrValue, models_library.services.Se...

Results (6.66s):
     995 passed
       1 failed
```

⚗️ Future extensions might:
- include pull artifacts as a ``fixture`` using special arguments in pytest_simcore plugin CLI (e.g. ``--test-data-folder``, ``--registry-url``, ``--registry-user``, ``registry-password``)
-  add a new job in e2e to run against deployed data?

## Related issue/s

- Closes #3517 
- Code in ``packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker_registry.py`` could be a starting point for https://github.com/ITISFoundation/osparc-simcore/issues/2165

## How to test

```cmd
make devenv
source .venv/bin/activate

# create env file
make .env

# install library
cd packages/models-library
make install-dev

#  pull tests metadata from master registry (default in .env)
make pull_test_data

# run diagnostics on test-data
pytest --vv -m diagnostics
```

### CODE REVIEW

File: packages/models-library/tests/data/metadata-ascent-1.3.2.json [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          "Human"
+        ]
+      }
+    },
+    "input_2": {
+      "displayOrder": 2,
+      "label": "Cuff geometry",
+      "description": "45\u00b0 contact wrap monopolar cuff or 360\u00b0 contact wrap monopolar cuff or 45\u00b0 contact wrap, bipolar cuff or 360\u00b0 contact wrap, bipolar cuff \n",
+      "type": "ref_contentSchema",
+      "contentSchema": {
```
Comment (sanderegg) [inline packages/models-library/tests/data/metadata-ascent-1.3.2.json:31]: So I have a question here, why are all the inputs contentSchema and not the outputs? or else why are there no "legacy" types in the inputs and only "legacy" types in the outputs?

File: services/web/server/setup.cfg [hunk 2]
```diff
@@ -14,6 +14,7 @@ commit_args = --no-verify
 [tool:pytest]
 addopts = --strict-markers
 asyncio_mode = auto
-markers = 
+markers =
 	slow: marks tests as slow (deselect with '-m "not slow"')
 	acceptance_test: "marks tests as 'acceptance tests' i.e. does the system do what the user expects? Typically those are workflows."
+  diagnostics: "can be used to run diagnostics against deployed data (e.g. database, registry etc)"
```
Comment (sanderegg) [inline services/web/server/setup.cfg:20]: why is this in the webserver setup file at all?

**Review by sanderegg — APPROVED** [review]
ok, now I'm curious to see this at work.

File: packages/models-library/tests/data/metadata-ascent-1.3.2.json (hunk 1 above)
Comment (pcrespov) [reply packages/models-library/tests/data/metadata-ascent-1.3.2.json:31]: the outputs in this example are all files. For the moment, the file still uses the "legacy" type.

File: packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker_registry.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+RepoName = str
+RepoTag = str
+
+
+class Registry:
+    # SEE https://docs.docker.com/registry/spec/api
+    # SEE  https://github.com/moby/moby/issues/9015
+
+    def __init__(self, **data):
+        data.setdefault("url", f'https://{os.environ.get("REGISTRY_URL")}')
```
Comment (mrnicegyu11) [inline packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker_registry.py:32]: minor feature request: Allow `http` as well

File: packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker_registry.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        return digest
+
+    def check_manifest(self, repo_name: RepoName, repo_reference: str) -> bool:
+        r = httpx.head(
+            f"{self.data.url}/v2/{repo_name}/manifests/{repo_reference}",
+            auth=self.data.auth,
+        )
+        if r.status_code == 400:
+            return False
+        # some other error?
```
Comment (mrnicegyu11) [inline packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker_registry.py:84]: minor: make this comment/hint a bit more verbose

**Review by mrnicegyu11 — APPROVED** [review]
lgtm thanks!

**Pull request merged**